api/client/src/intTest/java/org/projectnessie/client/auth/oauth2/ITOAuth2Client.java

```diff
@@ -162,6 +164,33 @@ void testOAuth2ClientInitialRefreshToken() {
     }
   }
 
+  @Test
+  void testOAuth2ClientIdle() throws InterruptedException {
```

Comment:
The test is admittedly a little lame, and takes 15 seconds. I wonder if it's necessary to keep it around. We could disable it maybe and only run when necessary.

Reply:
The idea of injecting a Clock is nice, I will look into that. However, it will imho help mostly with unit tests. Here, just the clock won't be enough because we also need to control when the executor is going to run the tasks, and even worse, we can't really control the token expiration times that Keycloak creates.

Reply:
I will try however to "shrink" the test execution time by making the overall delays more aggressive.

Reply:
The unit test `TestOAuth2Client#testBackgroundRefresh`, which mocks both the executor + keycloak, now checks the sleeping flag.

The only extra thing we test here is the executor shrinking its core thread pool to zero.

I managed to reduce this test to ~5 secs btw.

Otherwise yes we could mock Keycloak but that would lead to something like Quarkus `OidcWiremockResource` which seems a lot of work for little gain, wdyt?

Reply:
Could we mock at the OAuth2 HTTP/REST level?... e.g. delegating `fetchNewTokens()` and similar methods to an impl. controlled by the test?

Reply:
That's what `TestOAuth2Client` does.

Reply:
So can we do the sleep test there?

Reply:
But... we already do: 

https://github.com/projectnessie/nessie/blob/a77c21ed9ef70d607834401885d738b59939af4d/api/client/src/test/java/org/projectnessie/client/auth/oauth2/TestOAuth2Client.java#L122-L130

Is there anything else you would like to test in that test class?

Reply:
Hmmm.... I must have missed it or got confused :facepalm: